instrumentation.go

```diff
@@ -401,6 +406,17 @@ func WithEnv() InstrumentationOption {
 
 			err = errors.Join(err, e)
 		}
+		if s, err := newSamplerFromEnv(); err == nil {
```

Comment:
This is implicitly setting the default. That needs to be left for `newInstConfig` (see suggestion for `newSamplerFromEnv`).

Additionally, this is shadowing `err` from the outer scope. If `err` is not `nil` in this scope it will not be returned in the outer scope function. You may want to refactor to something like this:

```suggestion
		if s, e := newSamplerFromEnv(); e != nil {
			err = errors.Join(err, e)
		} else if s != nil {
```